receiver/prometheusreceiver/internal/metricsbuilder.go

```diff
@@ -93,29 +93,40 @@ func (b *metricBuilder) AddDataPoint(ls labels.Labels, t int64, v float64) error
 		b.numTimeseries++
 		b.droppedTimeseries++
 		return errMetricNameNotFound
+
```

Comment:
What would happen if we had a separate case for metricName "up" here, in which we didn't return (which filters out the "up" metric)?  That seems simpler than filtering it out here, and adding a new internal metric above.